### PR TITLE
Fixing database for weekly scan.

### PR DESCRIPTION
**Pull request by mohammedzee1000**
Added uuid to weekly scan job.
Added database storage to weekly scan worker.
Updated scanner weekly scan path to include DB ops.

Signed-off-by: Mohammed Zeeshan Ahmed <mohammed.zee1000@gmail.com>

### CODE REVIEW

**Comment by mohammedzee1000** [comment]
@dharmit @navidshaikh @bamachrn fixing the weekly scan thing PTAL.

File: container_pipeline/workers/scan.py [hunk 1]
```diff
@@ -70,6 +70,8 @@ def handle_job(self, job):
         if self.job.get("weekly"):
             scanners_data["action"] = "notify_user"
             self.queue.put(json.dumps(scanners_data), 'master_tube')
+            self.init_next_phase_data('delivery')
```
Comment (navidshaikh) [inline container_pipeline/workers/scan.py]: `notify`

File: container_pipeline/workers/scan.py [hunk 2]
```diff
@@ -70,6 +70,8 @@ def handle_job(self, job):
...
+            self.logger.debug("Put job for delivery on master tube")
```
Comment (navidshaikh) [inline container_pipeline/workers/scan.py]: Even though we have limitation from db fields point of view, surely we dont have one for the log messages. :smile: 
```
self.logger.debug("Weekly scan for {project }is complete.")
```
should fit better here.

File: jenkinsbuilder/weekly-scan.py [hunk 3]
```diff
@@ -86,9 +91,23 @@
...
+            # Initializing Database entries
+            project, created = Project.objects.get_or_create(name=project_name)
+            build = Build.objects.create(uuid=job_uuid, project=project,
+                                         status='queued',
+                                         start_time=timezone.now(),
+                                         weekly_scan=True
+                                         )
+            scan_phase, created = BuildPhase.objects.get_or_create(
+                build=build, phase='scan')
+            scan_phase.status = 'queued'
```
Comment (navidshaikh) [inline jenkinsbuilder/weekly-scan.py]: I think we already set `status=queued` on line number 183 of this file ? We dont need to re-set it to same value unless changed. Just remove line number 109 ?

File: container_pipeline/models/pipeline.py [hunk 4]
```diff
@@ -66,6 +66,7 @@ class Build(models.Model):
         ('sent', 'Sent'),
         ('failed', 'Failed')
     )
+    weekly_scan = models.BooleanField(default=False)
```
Comment (navidshaikh) [inline container_pipeline/models/pipeline.py]: +1 for this.

**Comment by mohammedzee1000** [comment]
#dotests

**Comment by mohammedzee1000** [comment]
@dharmit do you want to review this, so we can merge :)

**Comment by dharmit** [comment]
> @dharmit do you want to review this, so we can merge :)

Code review's already done AFAICS. I'd like you to run a few weekly scans from your test setup so that we can be sure that it's working as expected.

**Comment by mohammedzee1000** [comment]
This PR is currently being tested. While i have had success triggers with emails, I want to ensure that i get a good sample size of the same. Code wise, it is now good.

**Comment by mohammedzee1000** [comment]
I have recieved a good sample size of scan success emails in  test setup. I think we should be good.

@bamachrn @navidshaikh

**Pull request merged**